File: Java/Problem_067.java
import java.io.*;
import java.util.*;

public class Euler67 {
    public static void main(String[] args) throws Exception {
        ArrayList<int[]> tri = new ArrayList<>();

        try (BufferedReader br = new BufferedReader(new FileReader("triangle.txt"))) {
            String line;
            while ((line = br.readLine()) != null) {
                line = line.trim();
                if (line.isEmpty()) continue;
                String[] parts = line.split("\\s+");
                int[] row = new int[parts.length];
                for (int i = 0; i < parts.length; i++) row[i] = Integer.parseInt(parts[i]);
                tri.add(row);
            }
        }

        int[] dp = tri.get(tri.size() - 1).clone();
        for (int r = tri.size() - 2; r >= 0; r--) {
            int[] row = tri.get(r);
            for (int c = 0; c <= r; c++) {
                dp[c] = row[c] + Math.max(dp[c], dp[c + 1]);
            }
        }

        System.out.println(dp[0]); // 7273 
    }
}

File: Python/Problem_067.py
def euler_67(path: str = "triangle.txt") -> int:
    tri = []
    with open(path, "r", encoding="utf-8") as f:
        for line in f:
            line = line.strip()
            if line:
                tri.append(list(map(int, line.split())))

    dp = tri[-1][:]  # copy last row
    for r in range(len(tri) - 2, -1, -1):
        for c in range(r + 1):
            dp[c] = tri[r][c] + (dp[c] if dp[c] > dp[c + 1] else dp[c + 1])
    return dp[0]

if __name__ == "__main__":
    print(euler_67())  # 7273 
